fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java

```diff
@@ -191,6 +196,29 @@ public List<Pair<Long, Expression>> getFilters() {
         return filters;
     }
 
+    public void putFilterJoinType(Expression filter, JoinType joinType) {
+        filterJoinType.put(filter, joinType);
+    }
+
+    /**
+     * find out whether filters can match original joinType
+     * @param filters filters needs to put on this join
+     * @param joinType join type computed by join constraint
+     * @return can filters matched
+     */
+    public boolean isFilterJoinTypeMatched(List<Expression> filters, JoinType joinType) {
+        for (Expression filter : filters) {
```

Comment:
Pls confirm the definition of filter and join condition.

Reply:
done